### ISSUE TITLE
Airdrop contract(s) using SignatureBouncer

### ISSUE THREAD

**Issue by Shrugs** [OpenZeppelin/openzeppelin-solidity#928]
## 🎉 Description

It'd be cool if we had a simple `Airdrop` contract that used `SignatureBouncer.sol` to issue tokens (with emission variants like the crowdsale's Mintable/Allowance types) to those with valid signatures from a server.

Bonus points for a nodejs script that runs across a list of addresses and generates signatures for each of them.

More bonus points for a simple rails/express based api endpoint for generating these signatures which follows a flow like:

1. user signs some challenge data using metamask/hardware-wallet/whatever
2. user submits that signature to the server
3. server recovers the address from the signature
4. server checks to see if address is in off-chain whitelist
5. if so, server signs a SignatureBouncer hash allowing the address to call the `Airdrop` contract
6. user can then submit that signature to the airdrop contract and receive their tokens

- [ ] 🐛 This is a bug report.
- [x] 📈 This is a feature request.

**Comment by shrugs** [comment]
Yeah. I expect a lot of contracts will want to have single-use signatures. Do you think it would be more gas efficient to keep track of the `nonce` per-sender (with perhaps a range check like `nonce must be less than 1` for single-use) and then sign signatures with a nonce on the backend or would it be better to just do a mapping of `bytes32 -> boolean` to keep track of whether or not the signature has been used before. Probably the nonce version is better?

**Comment by SylTi** [comment]
Not sure, the nonce version is probably more optimized but harder to code.
The nonce in my example was mainly to allow multiple issuances of the same amount to the same address
